src/main/java/org/spongepowered/api/event/Event.java

```diff
@@ -25,23 +25,33 @@
 package org.spongepowered.api.event;
 
 import org.spongepowered.api.event.cause.Cause;
+import org.spongepowered.api.event.cause.EventContext;
 import org.spongepowered.api.event.impl.AbstractEvent;
 import org.spongepowered.api.eventgencore.annotation.AbsoluteSortPosition;
 import org.spongepowered.api.eventgencore.annotation.ImplementedBy;
 
 /**
  * An event called within Sponge.
  *
- * <p>This is a marker interface, which must be implemented
- * by any event used with the Sponge event bus.</p>
+ * <p>This is a marker interface, which must be implemented by any event used
+ * with the Sponge event bus.</p>
  */
 @ImplementedBy(value = AbstractEvent.class, priority = Integer.MIN_VALUE)
 public interface Event {
+
     /**
      * Get the cause for the event.
      *
-     * @return The last cause
+     * @return The cause
      */
     @AbsoluteSortPosition(0)
     Cause getCause();
+
+    default Object getSource() {
```

Comment:
A marker interface would just restrict what could be in the cause while providing nothing of value.